// Copyright 2019 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// The option proto for the TfLiteTensorsToClassificationCalculator.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";

message TfLiteTensorsToClassificationCalculatorOptions {
  extend .mediapipe.CalculatorOptions {
    optional TfLiteTensorsToClassificationCalculatorOptions ext = 266399463;
  }

  // Score threshold for preserving the class.
  optional float min_score_threshold = 1;
  // Number of highest scoring labels to output.  If top_k is not positive then
  // all labels are used.
  optional int32 top_k = 2;
  // Path to a label map file for getting the actual name of class ids.
  optional string label_map_path = 3;
  // Whether the input is a single float for binary classification.
  // When true, only a single float is expected in the input tensor and the
  // label map, if provided, is expected to have exactly two labels.
  // The single score(float) represent the probability of first label, and
  // 1 - score is the probabilility of the second label.
  optional bool binary_classification = 4;
}
